File: alternatenumbermethod1.py
n=int(input("Enter a number:"))#accepted number 
t=n#dummy variable 1
r=n#dummy variable 2
ctr=0#counter for digits in the number
a=[]#array to store digits of n
i=0#loop counter
sumeven=0#sum of digits in even position
sumodd=0#sum of digits in odd position
while n!=0:
  n=n//10
  ctr=ctr+1
if(ctr==1):
  print('{} is not an alternate number'.format(r))
else:
  
  for i in range(0,ctr,1):
    x=int(t%10)
    a.append(x);
    t=t/10
  for i in range(0,ctr,1):
    if i%2==0:
      sumeven+=a[i]
    else:
      sumodd+=a[i]
  if sumeven==sumodd:
    print("{}  is an alternate number".format(r))
  else:
   print("{}  is not an alternate number".format(r)) 

File: alternatenumbermethod1.java
import java.util.*;
class AlternateNumber
{
    Scanner ob=new Scanner(System.in);
     void main()
    {
        int n;//accepted number 
        int t;//dummy variable 1
        int r;//dummy variable 2
        int i;//loop counter
        int ctr=0;//counter for digits in the number
        int a[]=new int[100];//array to store digits of n
        int sumeven=0;//sum of digits in even position
        int sumodd=0;//sum of digits in odd position
        System.out.print("Enter a number:");
        n=ob.nextInt();
        t=n;
        r=n;
        while(n!=0)
        {
            ctr++;
            n=n/10;
            
        }
        if(ctr==1)
        {
            System.out.println("\n Alternate number operations cannot be performed on single digits");
        }
        else
        {
            for(i=0;i<ctr;i++)
            {
                a[i]=t%10;
                t=t/10;
                
            }
            for(i=0;i<ctr;i++)
            {
                if(i%2==0)
                sumeven+=a[i];
                else
                sumodd+=a[i];
            }
            if(sumeven==sumodd)
            System.out.println(r+"  is an alternate number");
            else
            System.out.println(r+"  is not an alternate number");
        }
        
        
        
        
    }
}